### PR TITLE
[DOC]: Move build instructions to dev guide

### PR DESCRIPTION
**Pull request by jarmak-nv**
## Description
Contributes to #908 and unblocks #977 

This PR moves the existing build from source instructions to the developer docs page. It does not make any changes to the instructions.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cuspatial/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by thomcom — APPROVED** [review]

File: docs/source/developer_guide/development_environment.md [hunk 1]
```diff
@@ -12,5 +12,50 @@ dependencies and RAPIDS libraries automatically fetched and correctly versioned.
...
+export `CUSPATIAL_HOME=$(pwd)/cuspatial` && \
+git clone https://github.com/rapidsai/cuspatial.git $CUSPATIAL_HOME
+```
+### Install dependencies
+
+1. `export CUSPATIAL_HOME=$(pwd)/cuspatial`
+2. clone the cuSpatial repo
+
+```shell
+conda env update --file conda/environments/all_cuda-115_arch-x86_64.yaml
```
Comment (isVoid) [inline docs/source/developer_guide/development_environment.md:35]:
```suggestion
conda env update --file conda/environments/all_cuda-118_arch-x86_64.yaml
```

Comment (jarmak-nv) [reply docs/source/developer_guide/development_environment.md]: Included this update on the new build.md file!

**Review by isVoid — COMMENTED** [review]
Sorry for the late review. Can you also add an item in the toctree of `developer_guide/index.md`, `build`?

**Pull request merged**